File: RosTopic/src/package/src/subscriber_node.py
#!/usr/bin/env python

import rospy
from package.msg import topic

def callback(msg):
    rospy.loginfo("PESAN:")
    rospy.loginfo("Nama= %s", msg.nama)
    rospy.loginfo("Usia= %d", msg.usia)
    rospy.loginfo("Tinggi= %f", msg.tinggi)
    rospy.loginfo("Kehadiran= %s\n", "hadir" if msg.kehadiran else "tidak hadir")

def subscriber_node_py():
    rospy.init_node('subscriber_node_py', anonymous=True)

    rospy.Subscriber('py_topic', topic, callback)

    rospy.spin()

if __name__ == '__main__':
    subscriber_node_py()

File: TugasModul4/RosTopic/src/package/src/subscriber_node.cpp
#include "ros/ros.h"
#include "package/topic.h"

void messageCallback(const package::topic::ConstPtr &msg) {

    ROS_INFO("PESAN : ");
    ROS_INFO("Nama = %s", msg->nama.c_str());
    ROS_INFO("Usia = %ld", msg->usia);
    ROS_INFO("Tinggi = %f", msg->tinggi);
    ROS_INFO("Kehadiran = %s\n", msg->kehadiran ? "hadir" : "tidak hadir");
}

int main(int argc, char **argv) {
    ros::init(argc, argv, "subscriber_node_cpp");

    ros::NodeHandle nh;

    ros::Subscriber data_subscriber = nh.subscribe("cpp_topic", 1000, messageCallback);

    ros::spin();

    return 0;
}
